### PR TITLE
Use MessageLite for grpc serialization

### PR DESCRIPTION
**Pull request by kwasimensah**
The full protobuf MessageType isn't needed for GRPC serialization, just it's base class MessageLite. This change allows using GRPC with protobufs generated from the lite runtime, allowing saving binary size on mobile devices.

### CODE REVIEW

File: include/grpcpp/impl/codegen/config_protobuf.h [hunk 1]
```diff
@@ -76,6 +78,7 @@ namespace grpc {
 namespace protobuf {
 
 typedef GRPC_CUSTOM_MESSAGE Message;
+typedef GRPC_CUSTOM_MESSAGE MessageLite;
```
Comment (yang-g) [inline include/grpcpp/impl/codegen/config_protobuf.h]: GRPC_CUSTOM_MESSAGELITE

Comment (kwasimensah) [reply include/grpcpp/impl/codegen/config_protobuf.h]: fixed

**Comment by yang-g** [comment]
Please fix formatting:
```
+ CLANG_FORMAT_ROOT=/var/local/git/grpc
+ tools/dockerfile/grpc_clang_format/clang_format_all_the_things.sh
--- /var/local/git/grpc/include/grpcpp/impl/codegen/proto_utils.h	2019-05-21 06:11:52.823736792 +0000
+++ /tmp/tmp.Foc2cknA7f	2019-05-21 06:14:23.335736792 +0000
@@ -68,7 +68,8 @@
 // BufferReader must be a subclass of ::protobuf::io::ZeroCopyInputStream.
 template <class ProtoBufferReader, class T>
-Status GenericDeserialize(ByteBuffer* buffer, grpc::protobuf::MessageLite* msg) {
+Status GenericDeserialize(ByteBuffer* buffer,
+                          grpc::protobuf::MessageLite* msg) {
   static_assert(std::is_base_of<protobuf::io::ZeroCopyInputStream,
                                 ProtoBufferReader>::value,
                 "ProtoBufferReader must be a subclass of "
@@ -102,15 +103,17 @@
 // objects and grpc_byte_buffers. More information about SerializationTraits can
 // be found in include/grpcpp/impl/codegen/serialization_traits.h.
 template <class T>
-class SerializationTraits<T, typename std::enable_if<std::is_base_of<
-                                 grpc::protobuf::MessageLite, T>::value>::type> {
+class SerializationTraits<
+    T, typename std::enable_if<
+           std::is_base_of<grpc::protobuf::MessageLite, T>::value>::type> {
  public:
-  static Status Serialize(const grpc::protobuf::MessageLite& msg, ByteBuffer* bb,
-                          bool* own_buffer) {
+  static Status Serialize(const grpc::protobuf::MessageLite& msg,
+                          ByteBuffer* bb, bool* own_buffer) {
     return GenericSerialize<ProtoBufferWriter, T>(msg, bb, own_buffer);
   }
-  static Status Deserialize(ByteBuffer* buffer, grpc::protobuf::MessageLite* msg) {
+  static Status Deserialize(ByteBuffer* buffer,
+                            grpc::protobuf::MessageLite* msg) {
     return GenericDeserialize<ProtoBufferReader, T>(buffer, msg);
   }
 };
```

**Comment by yang-g** [comment]
And squash commits..

**Pull request merged**